frontend/src/components/auth/ResetPassword.tsx

```diff
@@ -2,11 +2,35 @@ import React, { useContext } from "react";
 import authAPIClient from "../../APIClients/AuthAPIClient";
 import AuthContext from "../../contexts/AuthContext";
 
-const ResetPassword = () => {
+interface ResetPasswordProps {
+  email?: string;
+}
+
+const ResetPassword = ({ email }: ResetPasswordProps) => {
   const { authenticatedUser } = useContext(AuthContext);
 
+  const isRealEmailAvailable = (): boolean => {
+    const index = email!.indexOf("@");
+
+    if (index === -1 || email!.slice(index).indexOf(".") === -1) {
```

Comment:
nit: I suggest using a regex for emails https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript?page=1&tab=votes#tab-top